### PR TITLE
Fixes #566

### LINKED ISSUE #566: Cannot seed datetime generator
The faker lib allows you to seed Python's `random` instance, and the expected result is that all faker calls will have deterministic results every time the code is run. But the datetime generator relies on the `unix_time()` method, which drifts based on the current time:
```
@classmethod
def unix_time(cls):
    """
    Get a timestamp between January 1, 1970 and now
    :example 1061306726
    """
    return random.randint(0, int(time()))
```
The usage of time() as the end date means that datetime results change over time, even if random has been seeded.

I suggest making these methods accept an optional `end_datetime` (maybe `start_datetime` as well) parameter that would let the caller set the range of acceptable datetimes. That would allow users seeding random to get deterministic results.

### PR DESCRIPTION
**Pull request by kungfu71186**
Fixes #566, added end_datetime to all methods to provide consistent seeding when needed.

### CODE REVIEW

File: faker/providers/date_time/__init__.py [hunk 1]
```diff
@@ -289,46 +292,53 @@ def date_time_ad(self, tzinfo=None):
...
+    
+    @classmethod
+    def _parse_end_datetime(self, value):
+        if value is None:
+            return int(time())
+
+        return self._parse_date_time(value)
 
     @classmethod
-    def _parse_date_string(cls, value):
```
Comment (fcurella) [inline faker/providers/date_time/__init__.py]: could you leave the first argument of `classmethod`s as `cls`?

File: faker/providers/date_time/__init__.py [hunk 2]
```diff
@@ -246,21 +246,23 @@ class Provider(BaseProvider):
 
     regex = re.compile(timedelta_pattern)
 
-    def unix_time(self):
+    def unix_time(self, end_datetime = None):
```
Comment (fcurella) [inline faker/providers/date_time/__init__.py]: no space around the `=` in arguments, as per PEP8

File: faker/providers/date_time/__init__.py [hunk 3]
```diff
@@ -289,46 +292,53 @@ def date_time_ad(self, tzinfo=None):
...
+    def time_object(self, end_datetime = None):
         """
         Get a time object
         :example datetime.time(15, 56, 56, 772876)
         """
-        return self.date_time().time()
+        return self.date_time(end_datetime=end_datetime).time()
+    
+    @classmethod
+    def _parse_end_datetime(self, value):
```
Comment (fcurella) [inline faker/providers/date_time/__init__.py]: `@classmethod`s shuold use `cls` instead of `self`

Comment (kungfu71186) [reply faker/providers/date_time/__init__.py]: Yeah, I thought I had changed it back after I realized this. Ill take care of it

**Comment by fcurella** [comment]
Thank you! ✨

**Pull request merged**